types/module/module.go

```diff
@@ -494,6 +494,7 @@ func (m *Manager) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) abci.R
 // child context with an event manager to aggregate events emitted from all
 // modules.
 func (m *Manager) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) abci.ResponseEndBlock {
+	// Reset the EventManager, preserving any existing event history.
```

Comment:
```suggestion
	// [AGORIC] Reset the EventManager, preserving any existing event history.
```
Also, a readability suggestion:
```go
	eventHistory := []abci.Event{}
	if oldEm := ctx.EventManager(); oldEm != nil {
		eventHistory = oldEm.GetABCIEventHistory()
	}
	em := sdk.NewEventManagerWithHistory(eventHistory)
	ctx = ctx.WithEventManager(em)
```

Reply:
Done - yes, that's better.